src/main/CommandHandler.cpp

```diff
@@ -629,7 +629,8 @@ CommandHandler::dropPeer(std::string const& params, std::string& retStr)
             auto peer = peers.find(n);
             if (peer != peers.end())
             {
-                peer->second->drop(ERR_MISC, "dropped by user");
+                peer->second->drop(ERR_MISC, "dropped by user",
```

Comment:
There are two places where we need to flush write queue before closing connection.
One is when we are dropping peer because of load, and we want to send list of peers to it - before dropping we need to really send that list.

The other one is for promotion to outbound. We need to fully send our hello message in response to other peers hello (because we will drop in a moment, as in case of promotion we are already connected to that peer), so it can see that we can handshake and set outbound value to true.

Reply:
you didn't really answer my question: yes we can have this special handling, my question is, why bother with the "no flush" version?

Reply:
So you wan't all drops do the flush? We can do that, that should not do any harm.

Reply:
yeah how master works on that front seems fine

Reply:
Current master drops messages for all cases except when dropping because of ERR_LOAD (to send peers).